lib/ui/form/form_screen.dart

```diff
@@ -1,44 +1,126 @@
+
+final _errorStreamProvider = StreamProvider.autoDispose<String?>(
+    (ref) => ref.watch(formViewModelProvider.notifier).error);
 
-  // TODO: Integrate with VM #23
-  final _getSurveyDetailsUseCase = getIt.get<GetSurveyDetailsUseCase>();
+class FormScreen extends ConsumerStatefulWidget {
+  final String surveyId;
 
-  FormScreen({
+  const FormScreen({
     Key? key,
     required this.surveyId,
   }) : super(key: key);
 
+  @override
+  FormScreenState createState() => FormScreenState();
+}
+
+class FormScreenState extends ConsumerState<FormScreen> {
+  final _pageController = PageController();
+
+  @override
+  void initState() {
+    super.initState();
+    ref.read(formViewModelProvider.notifier).loadSurveyDetails(widget.surveyId);
+  }
+
   @override
   Widget build(BuildContext context) {
-    // TODO: Integrate with VM #23
-    _getSurveyDetailsUseCase.call(
-      GetSurveyDetailsInput(
-        surveyId: surveyId,
-      ),
-    );
+    final surveyDetails = ref.watch(_surveyDetailsStreamProvider).value;
+    final errorMessage = ref.watch(_errorStreamProvider).value ?? "";
+    return ref.watch(formViewModelProvider).when(
+          init: () => _buildFormScreen(isLoading: true),
+          loading: () => _buildFormScreen(isLoading: true),
+          loadSurveyDetailsSuccess: () => _buildFormScreen(
+            surveyDetails: surveyDetails,
+          ),
+          loadSurveyDetailsError: () => _buildFormScreen(
+            errorMessage: errorMessage,
+          ),
+        );
+  }
+
+  Widget _buildFormScreen({
+    bool isLoading = false,
+    SurveyDetailsModel? surveyDetails,
+    String errorMessage = "",
+  }) {
+    if (errorMessage.isNotEmpty) showSnackBar(context, errorMessage);
     return Scaffold(
       backgroundColor: Colors.black,
-      body: SafeArea(
-        child: PageView.builder(
-          // TODO: Integrate item count from survey details #23
-          itemCount: 3,
-          controller: _pageController,
-          itemBuilder: (BuildContext context, int index) {
-            if (index == 0) {
-              return const FormSurveyDetailPage();
-            } else {
-              return const FormSurveyQuestionPage();
-            }
+      body: Stack(
+        children: [
+          if (surveyDetails != null) ...[
+            DimmedBackground(background: surveyDetails.coverImageUrl),
+            PageView.builder(
+              // TODO: Integrate item count from survey details #25
+              itemCount: 3,
+
+              controller: _pageController,
+              itemBuilder: (BuildContext context, int index) {
+                if (index == 0) {
+                  return FormSurveyDetailPage(
+                    surveyDetails: surveyDetails,
+                  );
+                } else {
+                  return const FormSurveyQuestionPage();
+                }
+              },
+            ),
+            _buildStartButton()
```

Comment:
Does this mean we will have to show this Start Button for all question pages? Should this button remain on the `FormSurveyDetailPage`?

Reply:
or are you planning to show/hide the corresponding button here by the `index`?

Reply:
> or are you planning to show/hide the corresponding button here by the index?

Yes, correct. I've integrated this in the [next PR](https://github.com/nimblehq/ic-flutter-taher-toby/pull/71/files#diff-74190efa1f96d56465cdfdd4ed8a7df6402195aab6a35890f6b57d70027df792R107-R110) 🚀 

What do you think of this approach? 🙏

Reply:
@Tuubz That's fine 👍  Putting the button on each page will produce the Button that will be moving with the page when scrolling the pager index with animation. Putting the button outside of and on the page will not produce that result. Just a small different thing here, it's up to you to decide which one is the best, even making changes in the following PRs.